src/resolver.h

```diff
@@ -1 +1,6 @@
+#include "types.h"
+
 char* resolve(char*);
+
+header* build_header();
+void reset_counter();
```

Comment:
Why does the domain export these functions as part of resolver? Is the idea to allow other modules to build_header() through the resolver and use it for other functions? Similar question for reset_counter.